tanner/emulators/base.py

```diff
@@ -23,57 +15,66 @@ def __init__(self, base_dir, db_name, loop=None):
             'sqli': sqli.SqliEmulator(db_name, base_dir),
             'cmd_exec': cmd_exec.CmdExecEmulator()
         }
+        self.get_emulators = ['sqli', 'rfi', 'lfi', 'xss', 'cmd_exec']
+        self.post_emulators = ['sqli', 'rfi', 'lfi', 'xss', 'cmd_exec']
 
-    async def handle_post(self, session, data):
+    def extract_get_data(self, path):
+        """
+        Return all the GET parameter
+        :param path: The URL path from which GET parameters are to be extracted
+        :return: A MultiDictProxy object containg name and value of parameters
+        """
+        path = urllib.parse.unquote(path)
+        encodings = [('&&', '%26%26'), (';', '%3B')] 
+        for value, encoded_value in encodings:
+            path = path.replace(value, encoded_value)
+        get_data = yarl.URL(path).query
+        return get_data
+
+    async def get_emulation_result(self, session, data, target_emulators):
+        """
+        Return emulation result for the vulnerabilty of highest order
+        :param session: Current active session
```

Comment:
Well, self-documented code is always the first choice. But the thing is we've been working with a code so we feel it is unnecessary, but when I saw the code first time, it is difficult to go through the whole method and figure out and you just need to know what the method takes input and how does output look like. So I thought of writing doc strings. If you think it isn't needed at this stage, we can remove them but there is no harm in keeping them.